errors.go

```diff
@@ -35,19 +35,34 @@ var (
 // type to indicate that some non-transient error occurred while processing
 // the message, e.g. panic, (de|en)-coding errors or invalid usage of context.
 type errProcessing struct {
-	wrapped error
+	partition int32
+	err       error
 }
 
 func (ec *errProcessing) Error() string {
-	return fmt.Sprintf("error processing message: %v", ec.wrapped)
+	return fmt.Sprintf("error processing message (partition=%d): %v", ec.partition, ec.err)
 }
-func (ec *errProcessing) Unwrap() error {
-	return ec.wrapped
+func newErrProcessing(partition int32, err error) error {
+	return &errProcessing{
+		partition: partition,
+		err:       err,
+	}
 }
 
-func newErrProcessing(err error) error {
-	return &errProcessing{
-		wrapped: err,
+// type to indicate that some non-transient error occurred while processing
+// the message, e.g. panic, (de|en)-coding errors or invalid usage of context.
```

Comment:
actually the comment is wrong in this place, because this describes a setup-error (which cannot be a codec error).
But I fixed it in the other place and rewrite this comment.